source/auth/tests/mongodb-oidc.rst

```diff
@@ -5,31 +5,30 @@ MongoDB OIDC
 Drivers MUST test the following scenarios:
 
 - ``Callback-Driven Auth``
-- ``AWS Automatic Auth``
 - ``Callback Validation``
-- ``Cached Credentials``
 - ``Speculative Authentication``
 - ``Reauthentication``
+- ``Separate Connections Avoid Extra Callback Calls``
 
 
 .. sectnum::
 
-Drivers MUST be able to authenticate using either authentication or provider
-name if there are multiple principals configured on the server.  Note that
-``directConnection=true`` and ``readPreference=secondaryPreferred`` are needed because the server is a secondary on a replica set, on port ``27018``.
+Drivers MUST be able to authenticate against a server configured with either one or two configured identity providers.
 
-Drivers will need to be able to internally query and clear the cached
-credentials to verify usage for testing purposes.  Clearing the cache
-means removing all data from the cache, including ``OIDCMechanismServerStep1``
-information.
+Note that typically the preconfigured Atlas Dev clusters are used for testing, in Evergreen and localy.  The URIs can be fetched
+from the ``drivers/oidc`` Secrets vault, see `vault instructions`_.  Use ``OIDC_ATLAS_URI_SINGLE`` for ``MONGODB_URI_SINGLE`` and
+``OIDC_ATLAS_URI_MULTI`` for ``OIDC_ATLAS_URI_MULTI``.
 
-Drivers MUST set the ``AWS_WEB_IDENTITY_TOKEN_FILE`` environment variable
-to the location of valid ``test_user1`` credentials at the beginning of each
-test, unless otherwise specified.
+If using local servers is preferred, using the `Local Testing`_ method,
+use ``mongodb://localhost/?authMechanism=MONGODB-OIDC`` for ``MONGODB_URI_SINGLE`` and
+``mongodb://localhost:27018/?authMechanism=MONGODB-OIDC&directConnection=true&readPreference=secondaryPreferred``
+for ``MONGODB_URI_MULTI`` because the other server is a secondary on a replica set, on port ``27018``.
 
-Unless otherwise specified, tests will use a URL
-of the form ``mongodb://localhost/?authMechanism=MONGODB-OIDC``.
+The default OIDC client used in the tests will be configured with ``MONGODB_URI_SINGLE`` and a valid request callback handler
```

Comment:
Should this be implemented with `CUSTOM_TOKEN_CALLBACK` and `PROVIDER_NAME=custom`?

Reply:
No, it will depend on where it is being run.  We'll want to use the same unified tests on Azure, AWS, and GCP, hence the placeholder.